### PR TITLE
MAINT: optimize.minimize_scalar: always acknowledge 'bounds' and failures due to infinite bounds

### LINKED ISSUE #10911: scipy.optimize.minimize_scalar does not automatically select method='bounded' if bounds are given
Hi all,
This is a usability issue more than anything else. In short: if I pass bounds != None to minimize_scalar, I *also* have to pass `method='bounded'`, which seem rather redundant.
Furthermore, `method='bounded'` doesn't seem to deal with half-open intervals nicely.

#### Reproducing code example:

```
import scipy.optimize as so
def f(x): return x**2  # minimum at f(0) = 0

ans1 = so.minimize_scalar(f, bounds=(1, 100))
print(ans1.x)  # 0

ans2 = so.minimize_scalar(f, bounds=(1, None), method='bounded')
```
```
scipy/optimize/optimize.py in _minimize_scalar_bounded(func, bounds, args, xatol, maxiter, disp, **unknown_options)
   1758         raise ValueError("Optimisation bounds must be scalars"
   1759                          " or array scalars.")
-> 1760     if x1 > x2:
   1761         raise ValueError("The lower bound exceeds the upper bound.")
   1762 

TypeError: '>' not supported between instances of 'float' and 'NoneType'
```

#### Scipy/Numpy/Python version information:

1.3.1 1.17.2 sys.version_info(major=3, minor=6, micro=8, releaselevel='final', serial=0)

Keep up the good work :)

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Closes gh-10911

#### What does this implement/fix?
`minimize_scalar` silently ignores provided `bounds` unless `method='bounded'` is also passed. It can fail without producing an error or unsuccessful status if a bound is not finite. This PR fixes these issues.

#### Additional information
This is technically backward incompatible because the default `method='brent'` suggested that the default method was strictly Brent's (unbounded) method, but I think there is some wiggle room because `method='bounded'` is also said to use be a Brent method. When I've used `minimizer_scalar` in the past, I've expected it to use the bounded Brent method if `bounds` are passed, and so did the OP of gh-10911. I think it is safe to send an email to the mailing list to see if there are concerns, and if not, document the backward incompatibility in the release notes (without deprecation).

### CODE REVIEW

**Review by tupui — APPROVED** [review]
Thanks @mdhaber, LGTM.

To me it is really a bug fix as we should not be able to change a parameter which is then ignored. This is an API issue that needs fixing and to me we don't need a deprecation cycle. 

Since you sent an email, I suppose we have to wait a bit for potential comments.

**Comment by mdhaber** [comment]
If others agree that it's a bug fix, we can backport it into 1.10. If we don't hear otherwise, would you like to merge and apply the backport-candidate label on 12/12?

**Comment by mdhaber** [comment]
We gave it two weeks instead of one. Let's go for it.

**Review by tylerjereddy — APPROVED** [review]
Ok, you have a core developer approval for an effective bug-fix + backport and it sounds like you consulted the mailing list. I may squash merge this to facilitate my backporting duties.

**Pull request merged**
